content/blog/2022-01-15-moving-local-experiments-to-the-cloud-with-iterative-terraform-provider.md

```diff
@@ -0,0 +1,273 @@
+
+- _name_ (**required**): this is a name that will be used to set up the cloud
+  resources, and it does not have to be the same as the `iterative_task` name.
+  However, try to make it unique, it shouldn’t be in conflict with names of
+  existing cloud resources.
+- _cloud_ (**required**): cloud provider to run the task on. This can be `aws`,
+  `gcp`, `az`, or `k8s`.
+- _region_: you can choose the region where the compute resources should be
+  allocated.
+- _machine_: if you know the exact kind of machine that you'd like to use, you
+  can specify it here. Alternatively, the Iterative Provider offers some common
+  machine types which are roughly the same for all supported clouds. For
+  example,
+  [l+k80](https://registry.terraform.io/providers/iterative/iterative/latest/docs/resources/task#l+k80)
+  stands for a "Large, with (at least) 12 CPU cores, 112 GB of RAM and 2 GPU
+  devices".
+- _workdir_: Specify a directory on your local machine relative to your project
+  folder which you would like to sync with the remote machine. This way you can
+  share your whole project or parts of it with a remote machine. In my example,
+  I am sharing the whole project.
+- _script_ (**required)**: This is where the Iterative Provider's magic happens,
+  i.e. this is where we define a script that should be run on a provisioned
+  machine.
+
+Take a look at the script. In the simplest scenario, all we need to do on a new
+machine to run the training script is to set up the Python environment with
+required libraries. If you simply want to train your model on a machine with
+more memory, this may be enough. However, if you want your training code to
+leverage GPUs, the script will look a bit different.
+
+## Training with GPU
+
+There are several ways you can leverage GPU devices on a remote machine. You can
+install all the required drivers and dependencies "manually" via a script, you
+can use an existing Docker image or build your own, or, in the case of AWS, you
+can take advantage of an
+[existing pre-configured Deep Learning AMI](https://docs.aws.amazon.com/dlami/latest/devguide/options.html).
+In this tutorial, we'll use an AMI, as it is the quickest way to start using
+GPUs on an AWS EC2 machine.
+
+```hcl
+resource "iterative_task" "tpi-examples-gpu" {
+    name      = "ami-gpu-example"
+    cloud     = "aws"
+    region    = "us-east-2"
+    machine   = "l+k80"
+    disk_size = "130"
+    image     = "ubuntu@898082745236:x86_64:Deep Learning AMI (Ubuntu 18.04) Version 54.0"
+
+    workdir { input = "." }
+    script = <<-END
+    #!/bin/bash
+    pip3 install -r requirements.txt
+    python3 src/train.py
+    END
+}
+```
+
+To use a Deep Learning AMI, you need to configure it via the `image` argument
+that has the following structure: `{user}@{owner}:{architecture}:{name}`. You
+can learn more about AMIs in the official
+[AWS documentation](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/AMIs.html).
```

Comment:
Minor: this paragraph could make more sense before the code block.

Reply:
```suggestion
```